src/pages/domain/VerifyDomainPage.tsx

```diff
@@ -0,0 +1,167 @@
+import {Str} from 'expensify-common';
+import React, {useEffect} from 'react';
+import type {PropsWithChildren} from 'react';
+import {View} from 'react-native';
+import Button from '@components/Button';
+import CopyableTextField from '@components/Domain/CopyableTextField';
+import FormAlertWithSubmitButton from '@components/FormAlertWithSubmitButton';
+import FormHelpMessage from '@components/FormHelpMessage';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Icon from '@components/Icon';
+import {Exclamation} from '@components/Icon/Expensicons';
+import RenderHTML from '@components/RenderHTML';
+import ScreenWrapper from '@components/ScreenWrapper';
+import ScrollView from '@components/ScrollView';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useOnyx from '@hooks/useOnyx';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {getDomainValidationCode, resetDomainValidationError, validateDomain} from '@libs/actions/Domain';
+import {getLatestErrorMessage} from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {PlatformStackScreenProps} from '@libs/Navigation/PlatformStackNavigation/types';
+import type {WorkspacesDomainModalNavigatorParamList} from '@libs/Navigation/types';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+function OrderedListRow({index, children}: PropsWithChildren<{index: number}>) {
+    const styles = useThemeStyles();
+    return (
+        <View style={[styles.flexRow, styles.flex1]}>
+            <Text style={styles.pr1}>{index}.</Text>
+            {children}
+        </View>
+    );
+}
+
+type VerifyDomainPageProps = PlatformStackScreenProps<WorkspacesDomainModalNavigatorParamList, typeof SCREENS.WORKSPACES_VERIFY_DOMAIN>;
+
+function VerifyDomainPage({route}: VerifyDomainPageProps) {
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+
+    const accountID = route.params.accountID;
+    const [domain] = useOnyx(`${ONYXKEYS.COLLECTION.DOMAIN}${accountID}`, {canBeMissing: true});
+    const domainName = domain ? Str.extractEmailDomain(domain.email) : '';
+    const {isOffline} = useNetwork();
+
+    const doesDomainExist = !!domain;
+
+    useEffect(() => {
+        if (!domain?.validated) {
```

Comment:
in the DomainVerifiedPage.tsx component you are also checking if the !!domain is truthy here, should we add it here too or remove it in that component if its not necessary